lms/migrations/versions/3273c40d33a8_events.py

```diff
@@ -0,0 +1,137 @@
+        sa.Column("grouping_id", sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["application_instance_id"],
+            ["application_instances.id"],
+            name=op.f("fk__event__application_instance_id__application_instances"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["assignment_id"],
+            ["assignment.id"],
+            name=op.f("fk__event__assignment_id__assignment"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["course_id"],
+            ["grouping.id"],
+            name=op.f("fk__event__course_id__grouping"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["grouping_id"],
+            ["grouping.id"],
+            name=op.f("fk__event__grouping_id__grouping"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["type_id"],
+            ["event_type.id"],
+            name=op.f("fk__event__type_id__event_type"),
+            ondelete="cascade",
+        ),
+        sa.PrimaryKeyConstraint("id", name=op.f("pk__event")),
+    )
+    op.create_index(op.f("ix__event_timestamp"), "event", ["timestamp"], unique=False)
+    op.create_table(
+        "event_data",
+        sa.Column("event_id", sa.Integer(), nullable=False),
+        sa.Column(
+            "extra",
+            postgresql.JSONB(astext_type=sa.Text()),
+            server_default=sa.text("'{}'::jsonb"),
+            nullable=False,
+        ),
+        sa.ForeignKeyConstraint(
+            ["event_id"],
+            ["event.id"],
+            name=op.f("fk__event_data__event_id__event"),
+            ondelete="cascade",
+        ),
+        sa.PrimaryKeyConstraint("event_id", name=op.f("pk__event_data")),
+    )
+    op.create_table(
+        "event_user",
+        sa.Column("id", sa.Integer(), autoincrement=True, nullable=False),
+        sa.Column("event_id", sa.Integer(), nullable=False),
+        sa.Column("user_id", sa.Integer(), nullable=False),
+        sa.Column("lti_role_id", sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(
+            ["event_id"],
+            ["event.id"],
+            name=op.f("fk__event_user__event_id__event"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["lti_role_id"],
+            ["lti_role.id"],
+            name=op.f("fk__event_user__lti_role_id__lti_role"),
+            ondelete="cascade",
+        ),
+        sa.ForeignKeyConstraint(
+            ["user_id"],
+            ["user.id"],
+            name=op.f("fk__event_user__user_id__user"),
+            ondelete="cascade",
+        ),
+        sa.PrimaryKeyConstraint("id", "event_id", name=op.f("pk__event_user")),
+        sa.UniqueConstraint(
+            "user_id", "lti_role_id", name=op.f("uq__event_user__user_id")
+        ),
+    )
+
```

Comment:
Should I backfill the new table with the old one here?

If we want to do it I reckon now is the best time.